### DISCUSSION TITLE
Custom workflow executing

### DISCUSSION THREAD

**Discussion by IgorRSU** [YetiForceCompany/YetiForceCRM#14962] [General] [answered]
Hi, a I have a problem - I`m creating Custom workflow for Contacts with executing Every time a record is saved, without filter conditions.
When I choose "Sum field values from related module fields" - it executing immediately (after record save), but when I choose  "Update related record" WF is executing after Cron Job Workflow job ends.
Is it ok? And can I make immediately execution for Update related record task?

**Comment by davide-alghi** [comment, marked as answer]
Hi @IgorRSU,
please take a look here

https://github.com/YetiForceCompany/YetiForceCRM/blob/a4ee5adc6d8826debc8d04e63cb96fca59d24aac/modules/com_vtiger_workflow/tasks/SumFieldFromDependent.php#L17

and here

https://github.com/YetiForceCompany/YetiForceCRM/blob/a4ee5adc6d8826debc8d04e63cb96fca59d24aac/modules/com_vtiger_workflow/tasks/VTUpdateRelatedFieldTask.php#L13

**Reply by mariuszkrzaczkowski** [reply]
this may take longer to save data, be aware of this

**Reply by IgorRSU** [reply]
I changed to true public $executeImmediately in VTUpdateRelatedFieldTask.php and nothing happened, field is updating after WF job ends. Any idea?
Thanks

**Reply by davide-alghi** [reply]
@IgorRSU,
it works.
Take into account that the ```$executeImmediately``` value is stored in database on workflow task creation.
So you have:
1. to move ```$executeImmediately``` to ```true``` value
2. to re-create the task in workflow (delete the existing one and re-create it)
